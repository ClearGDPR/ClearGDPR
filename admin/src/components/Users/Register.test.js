import React from 'react';
import { shallow, mount } from 'enzyme';

import RegisterForm, { Register } from './Register';

const setupShallow = propOverrides => {
  const props = Object.assign({
    onSubmit: jest.fn()
  }, propOverrides);
  const component = shallow(<Register {...props} />);

  return { props, component };
};

const setupMount = propOverrides => {
  const props = Object.assign({
    isLoading: false
  }, propOverrides);
  const component = mount(<RegisterForm {...props} />);

  return { props, component };
};

describe('(Component) Register', () => {
  it('should render correctly when no props provided', async () => {
    const { component } = setupShallow();
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when the prop isLoading is set to false', async () => {
    const { component } = setupShallow({ isLoading: false });
    expect(component).toMatchSnapshot();
  });

  it('should render correctly when the prop isLoading is set to true', async () => {
    const { component } = setupShallow({ isLoading: true });
    expect(component).toMatchSnapshot();
  });

  it('should validate password when inputs values change', async () => {
    // TODO: refactor
  });

  it('should submit when validation is ok', async () => {
    // TODO: refactor
  });

  it('should render correct props when touched', async () => {
    const { component } = setupShallow({
      isLoading: false,
      touched: { password: true }
    });

    const passwordInput = component.find('TextInput[field="password"]').at(0);

    expect(passwordInput.props().error).toBeFalsy();
  });

  it('should render correct props when touched and there are errors', async () => {
    const { component } = setupShallow({
      isLoading: false,
      touched: { password: true },
      errors: { password: 'Error message' }
    });

    const passwordInput = component.find('TextInput[field="password"]').at(0);

    expect(passwordInput.props().error).toEqual('Error message');
  });
});
