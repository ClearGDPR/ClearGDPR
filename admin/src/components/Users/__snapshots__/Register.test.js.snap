// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Register should render correctly when no props provided 1`] = `
<React.Fragment>
  <TextInput
    field="username"
    label="Username"
    placeholder="mick.bern"
    required={true}
  />
  <TextInput
    field="password"
    label="New password"
    placeholder="*********"
    type="password"
  />
  <Component
    className="btn"
    text="Register"
    type="submit"
  />
</React.Fragment>
`;

exports[`(Component) Register should render correctly when the prop isLoading is set to false 1`] = `
<React.Fragment>
  <TextInput
    field="username"
    label="Username"
    placeholder="mick.bern"
    required={true}
  />
  <TextInput
    field="password"
    label="New password"
    placeholder="*********"
    type="password"
  />
  <Component
    className="btn"
    text="Register"
    type="submit"
  />
</React.Fragment>
`;

exports[`(Component) Register should render correctly when the prop isLoading is set to true 1`] = `
<React.Fragment>
  <TextInput
    field="username"
    label="Username"
    placeholder="mick.bern"
    required={true}
  />
  <TextInput
    field="password"
    label="New password"
    placeholder="*********"
    type="password"
  />
  <Component
    className="btn"
    text="Register"
    type="submit"
  />
</React.Fragment>
`;
